op-batcher/batcher/service.go

```diff
@@ -164,37 +163,24 @@ func (bs *BatcherService) initRPCClients(ctx context.Context, cfg *CLIConfig) (o
 		ethUrls := strings.Split(cfg.L2EthRpc, ",")
 		provider, err := dial.NewActiveL2EndpointProvider(ctx, ethUrls, rollupUrls, cfg.ActiveSequencerCheckDuration, dial.DefaultDialTimeout, bs.Log)
 		if err != nil {
-			return nil, fmt.Errorf("failed to build active L2 endpoint provider: %w", err)
+			return fmt.Errorf("failed to build active L2 endpoint provider: %w", err)
 		}
 		endpointProvider = provider
-
-		// If we use an active endpoint provider AND throttling is enabled,
-		// we need to set up the callback to notify the driver any time we
-		// get a new active sequencer
-		if cfg.ThrottleThreshold > 0 {
-			activeSeqChanged := make(chan struct{}, 1)
-			opts = []DriverSetupOption{func(setup *DriverSetup) {
-				setup.ActiveSeqChanged = activeSeqChanged
-			}}
-			// callback to notify the driver of a new active sequencer
-			cb := func() {
-				select {
-				case activeSeqChanged <- struct{}{}:
-				default:
-				}
-			}
-			provider.SetOnActiveProviderChanged(cb)
-
+		if len(bs.ThrottlingEndpoints) == 0 {
+			bs.ThrottlingEndpoints = ethUrls
 		}
 	} else {
 		endpointProvider, err = dial.NewStaticL2EndpointProvider(ctx, bs.Log, cfg.L2EthRpc, cfg.RollupRpc)
 		if err != nil {
-			return nil, fmt.Errorf("failed to build static L2 endpoint provider: %w", err)
+			return fmt.Errorf("failed to build static L2 endpoint provider: %w", err)
+		}
+		if len(bs.ThrottlingEndpoints) == 0 {
+			bs.ThrottlingEndpoints = []string{cfg.L2EthRpc}
```

Comment:
This is possibly not the ideal place to do the fallback when there's an empty list. But it lead to the smallest diff and least amount of code duplication to do it here. 

It's possible we don't even need the separate list of throttling endpoints, in which case this code will simplify.

Reply:
I think we actually need to go a bit further here and compute the union of the `cfg.L2EthRpc` and `bs.ThrottlingEndpoints`. I think we should rename `bs.AdditionalThrottlingEndpoints` and document it clearly such that we don't need to duplicate the endpoints from the `l2-eth-rpc` flag.

Reply:
Wrote this comment before I saw yours. Agreed we can compute the union and use that. Adding the regular sequencers to the `AdditionalThrottlingEndpoints` list won't hurt, but it also won't be strictly necessary either as long as they are in the l2-eth-rpc list.

Reply:
Implemented it here [9ad69f2](https://github.com/ethereum-optimism/optimism/pull/15603/commits/9ad69f276e15c7684cea3c61dceae519be728ae9) and here [aec2c00](https://github.com/ethereum-optimism/optimism/pull/15603/commits/aec2c00a33b69746c415ba853820cc8efd01a1b0)